.github/workflows/spanner-emulator-pr-push.yml

```diff
@@ -0,0 +1,46 @@
+name: Spanner tests against emulator (Spanner changes only)
+
+on:
+  pull_request:
+    paths:
+    - '**[Ss]panner**'
+  push:
+    branches:
+      - master
+    paths:
+    - '**[Ss]panner**'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      DOTNET_NOLOGO: true
+    services:
+      emulator:
+        image: gcr.io/cloud-spanner-emulator/emulator:latest
+        ports:
+          - 9010:9010
+          - 9020:9020
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+        fetch-depth: 100
+
+    # The GitHub checkout action leaves the repo in a slightly awkward
+    # state. This tidies it up.
+    - name: Set up git branches
+      if: ${{ github.event_name == 'pull_request' }}
+      run: |
+        git checkout -b pr-head
+        git checkout -b master ${{ github.event.pull_request.base.sha }}
+        git checkout pr-head
+
+    - name: Spanner Emulator
+      env:
+          SPANNER_EMULATOR_HOST: localhost:9010
+          TEST_PROJECT: emulator-test-project
+      run: |
+        dotnet test ./apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests
```

Comment:
I think this will use the debug version - not sure whether it's worth specifying `-c Release`